### DISCUSSION TITLE
Can we pass JS event into func ?

### DISCUSSION THREAD

**Discussion by SJ22032003** [a-h/templ#678] [Q&A] [answered]
can we pass js events into go functions ?
like in javascript we can simple do this.
`<button onclick="myFunc(e)"> click </button>`

I am using templ func to run functions in my htmx request and its working fine, but I want to access the event and pass them on my  functions.

![Screenshot from 2024-04-09 15-56-07](https://github.com/a-h/templ/assets/85823986/3ead9e3a-c392-442f-9da4-99780773c486)

so something like getting the event from hx-on::before-request into `showModel`. so I can you know pass it into `model.ShowModel(event)` which is returning a Component Script, so I can use event.preventDefault()

I am open for other workarounds if this is not possible

**Comment by PaluMacil** [comment]
For frontend code to call Go functions you're going to have that function available via wasm and running in the browser separate from the server Go code with all the implications of it being two separate processes that you'll need to keep in mind, or you will need to make that function. Be something that calls the back end and links to that specific function on the backend, retrieves the answer and then uses it in the frontend. Both are possible, but javascript in the browser is running in a physically different location than your go code right now. So one of these two things would need to be implemented first, and there are enough ways someone might want this done that probably you wouldn't want in templ.

Something that might be confusing is that it looks like the reverse is possible. You can put go values in JavaScript using the templating available. That's because you are rendering the JavaScript with those values, but once that is sent to the front end, the values you got from the back end are already there and you would need to communicate to the back end to get more or different data. Reloading the page with different query parameters, the backend could send different data because it would be another round trip to the server, but it isn't actually go communicating with JavaScript, it's go templating JavaScript with some values in the JavaScript already printed out to be interpreted by the browser when it arrives there.

**Reply by SJ22032003** [reply, marked as answer]
Yes!!!! How could I be soo dumb, they are literally running on different environments. Ngl templ makes it feel like writing reactive code in the client. My bad.
